CRM/CiviGeometry/BAO/Geometry.php

```diff
@@ -87,57 +87,61 @@ public static function getCentroid($params) {
    */
   public static function contains($params) {
     $multipleResult = [];
-    $duleIntegerSQL = "SELECT ST_Contains(a.geometry, b.geometry)
+    $dualIntegerSQL = "SELECT a.id, ST_Contains(a.geometry, b.geometry) as contains_result
       FROM civigeometry_geometry a, civigeometry_geometry b
-      WHERE a.id = %1, and b.id = %2";
-    $singleIntergerSQL = "SELECT ST_Contains(geometry, GeomFromText(%1, 4326))
-      FROM civigeometry_geometry
-      WHERE id = %2";
+      WHERE b.id = %2";
+    $singleIntergerSQL = "SELECT id, ST_Contains(geometry, GeomFromText(%1, 4326)) as contains_result
+      FROM civigeometry_geometry";
     if ($params['geometry_a'] == 0) {
       $geometryParams = [
-        'is_active' => 1,
+        'is_archived' => 0,
         'options' => ['limit' => 0],
         'return' => ['id'],
       ];
       if (!empty($params['geometry_a_collection_id'])) {
         $geometryParams['collection_id'] = $params['geometry_a_collection_id'];
       }
-      $geometries = civicrm_api3('Geometry', 'get', $geometryParams);
-      foreach ($geometries['values'] as $geometry) {
-        if (is_numeric($params['geometry_b'])) {
-          $res = CRM_Core_DAO::singleValueQuery($duleIntegerSQL, [
-            1 => [$geometry['id'], 'Positive'],
-            2 => [$params['geometry_b'], 'Positive'],
-          ]);
-        }
-        else {
-          $res = CRM_Core_DAO::singleValueQuery($singleIntergerSQL, [
-            1 => [$params['geometry_b'], 'String'],
-            2 => [$geometry['id'], 'Positive'],
-          ]);
-        }
-        if (!empty($res)) {
-          $multipleResult[] = $geometry['id'];
+      $geometries = civicrm_api3('Geometry', 'get', $geometryParams)['values'];
+      $singleIntergerSQL .= " WHERE id IN (%2)";
+      $dualIntegerSQL .= "AND a.id IN (%1)";
```

Comment:
I think you're going to need an extra space before "AND" here.

Reply:
@andrew-cormick-dockery good pick up i extended a unit test in https://github.com/australiangreens/au.org.greens.civigeometry/pull/12/commits/c530d86541581cd6e7104b97492acaf18285bc05 which demonstrates the issue you raise here and then fixed it in https://github.com/australiangreens/au.org.greens.civigeometry/pull/12/commits/25e3bac79c14248ce39c276f81dee9298537724d

Reply:
Test failure demonstrated here https://buildkite.com/greens/au-dot-org-dot-greens-dot-civigeometry/builds/52#768d9204-f68d-4515-a385-359761b78eef/942-970 and passing here https://buildkite.com/greens/au-dot-org-dot-greens-dot-civigeometry/builds/53#5c4d0364-28bc-408d-b955-f9f3d0397451/769-795